satpy/tests/reader_tests/test_eum_base.py

```diff
@@ -101,6 +102,27 @@ def test_fun(self):
 
         self.assertEqual(recarray2dict(pat), expected)
 
+    def test_mpef_product_header(self):
+        """Test function for TestRecarray2Dict and mpef product header."""
+        mpef_header = np.dtype([
```

Comment:
I'm still a bit concerned that the datatype used for testing is a copy of the original datatype (`mpef_product_header`). I guess testing the conversion of the entire header would be the best solution, but that's probably beyond the scope of this PR :slightly_smiling_face:  Maybe subsetting the original is a good compromise?

```python
from satpy.readers.seviri_base import mpef_product_header
names = ['ImageLocation', 'GsicsCalMode', 'GsicsCalValidity', 'Padding', 'OffsetToData' 'Padding2']
mpef_header = np.dtype([(name, mpef_product_header.fields[name][0]) for name in names])
mph_string = np.array(..., dtype=mpef_header)
```